modelforge/tests/test_training.py

```diff
@@ -57,6 +53,34 @@ def test_train_with_lightning(model_name, dataset_name):
     assert type(model) is not None
 
 
+@pytest.mark.parametrize("model_name", ["ANI2x"])
+@pytest.mark.parametrize("dataset_name", _ImplementedDatasets.get_all_dataset_names())
+def test_loss(model_name, dataset_name):
+
+    from modelforge.dataset.dataset import DataModule
+
+    dm = DataModule(
+        name=dataset_name,
+        batch_size=512,
+        remove_self_energies=True,
+        for_unit_testing=True,
+    )
+    dm.prepare_data()
+    dm.setup()
+    model = NeuralNetworkPotentialFactory.create_nnp("inference", model_name)
+
+    from modelforge.train.training import EnergyAndForceLoss
+    import torch
+
+    loss = EnergyAndForceLoss(model)
+
+    r = loss.compute_loss(next(iter(dm.train_dataloader())))
+
+    loss = EnergyAndForceLoss(model, include_force=True)
```

Comment:
@chrisiacovella , can you please take a look at the test failure for the datasets in this test? It seems as if there are indexing errors for the datasets (both for the energy and the force loss)

Reply:
The errors for the energy loss for the different variants of the SPICE are caused by a mismatch between the elements covered by the ANI network and the elements present in the dataset. I think this is fine, but we should through a more explicit and understandable error than a `IndexError` (which obscures the underlying cause).

But I can't figure out the indexing error in the force loss.

Reply:
@wiederm I'm going to additional set up a dataset for spice that only includes the elements supported by ANI (both a "full" and "testing" dataset).  I'm trying to decide the best way to handle this. I think it might be easy to put a filtering option in the dataset curation, like I've done for addressing limiting the number of conformers for the training set generation.  I initially thought having this be a post processing step, which would be fine for the full datasets, but not for test datasets that are meant to have a specific number of conformers.

Reply:
that is a great idea!

The problem is only for ANI, though; none of the other potentials are limited to a set of elements. So I think it's also fine to just don't allow this combination of model and dataset.